### ISSUE TITLE
InSuperCall after LambdaLift

### ISSUE THREAD

**Issue by DarkDimius** [lampepfl/dotty#484]
```
class S[A](f: A => A, x: A) {
  Console.println(f(x));
}
class T[B](f: B => B, y: B) extends S((x: B) => f(x), y) {
}
```
after lambda lift 
```
class S extends Object {
    def <init>(f: Function1, x: Object): S = {
      super()
      Console.println(f.apply(x))
      ()
    }
  }
class T extends S {
    def <init>(f: Function1, y: Object): T = {
      super(closure(f | $anonfun$$init$$6:Function1), y)
      T.this.f = f
      ()
    }
    private val f: Function1
    private def $anonfun$$init$$6(f$2: Function1, x: Object): Object = f$2.apply(x)
  }
```

Where `$anonfun$$init$$6` still has `InSuperCall` flag set. If one tries to call `owner` on it, `module class <empty>` will be returned.

I guess lambdaLift should also reset `InSuperCall` flag when lifting methods.
@odersky WDYT?

**Comment by odersky** [comment]
Yes, I think this is right. I'll prepare a PR.

On Thu, Apr 16, 2015 at 5:03 PM, Dmitry Petrashko <notifications@github.com>
wrote:

> class S[A](f: A => A, x: A) {
>   Console.println(f(x));
> }
> class T[B](f: B => B, y: B) extends S((x: B) => f(x), y) {
> }
>
> after lambda lift
>
> class S extends Object {
>     def <init>(f: Function1, x: Object): S = {
>       super()
>       Console.println(f.apply(x))
>       ()
>     }
>   }
> class T extends S {
>     def <init>(f: Function1, y: Object): T = {
>       super(closure(f | $anonfun$$init$$6:Function1), y)
>       T.this.f = f
>       ()
>     }
>     private val f: Function1
>     private def $anonfun$$init$$6(f$2: Function1, x: Object): Object = f$2.apply(x)
>   }
>
> Where $anonfun$$init$$6 still has InSuperCall flag set. If one tries to
> call owner on it, module class <empty> will be returned.
>
> I guess lambdaLift should also reset InSuperCall flag when lifting
> methods.
> @odersky <https://github.com/odersky> WDYT?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lampepfl/dotty/issues/484>.
>



-- 
Martin Odersky
EPFL

**Comment by DarkDimius** [comment]
Implemented in #486
